### DISCUSSION TITLE
Getting System.Net.Quic is not supported on this platform on Linux

### DISCUSSION THREAD

**Discussion by kamranattari4192** [microsoft/msquic#3664] [Q&A] [answered]
Hi,
I am trying to run a .NET 7 application on CentOS7. It is always giving following error:
System.Net.Quic is not supported on this platform
I have installed libmsquic.
On another CentOS7 VM, the same code is working. What could be the issue on first VM. How can I diagnose the root cause of this issue?

**Comment by ManickaP** [comment]
You can turn on internal diagnostics for S.N.Quic: "Private.InternalDiagnostics.System.Net.Quic". For example, you can do that programmatically like this https://gist.github.com/ManickaP/864c3ee9c854db58db9ad6d54957d947, just change the `eventSource.Name` to "Private.InternalDiagnostics.System.Net.Quic" here https://gist.github.com/ManickaP/864c3ee9c854db58db9ad6d54957d947#file-http_test_logging-cs-L14.

If that doesn't show anything meaningful, the app cannot find and load libmsquic.so. So first step would be to locate it: `ldconfig -p | grep libmsquic`. Then you can check if it can be loaded: `ldd /usr/lib/libmsquic.so.2`.

In the end one of these cases will be the culprit:
- it's not actually installed in the system
- it's installed, but some of the dependencies are missing and/or are incompatible (there were some issues in this area)
- it's installed, but it's too old version that is incompatible with .NET 7

**Reply by nibanks** [reply]
@manickap this seems like something perfect to copy into your TSG.

**Comment by kamranattari4192** [comment]
Hi,

Thanks for your reply. I deleted my new CentOS 7 VM and tried with following steps: 
1. Installed new RHEL 7.8
2. Installed libmsquic
3. Installed dotnet 7
4. Run my test application that uses MsQuic, it worked.
5. However, when I run **ldconfig -p | grep libmsquic**, it shows nothing.
6. When I run **ldd /usr/lib/libmsquic.so.2**, it shows me this error **ldd: /usr/lib/libmsquic.so.2: No such file or directory**. But MsQuic is working fine.
7. Then I tried this **ldd /lib64/libmsquic.so.2**, it shows me followings;
linux-vdso.so.1 =>  (0x00007ffcaf37d000)
        libcrypto.so.1.1 => /lib64/libcrypto.so.1.1 (0x00007f2ba62ee000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007f2ba60ea000)
        libnuma.so.1 => /lib64/libnuma.so.1 (0x00007f2ba5ede000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f2ba5cc2000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f2ba58f4000)
        libz.so.1 => /lib64/libz.so.1 (0x00007f2ba56de000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f2ba67ca000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f2ba54c8000)
9. On this VM when I run **openssl version** it shows this _OpenSSL 1.0.2k-fips  26 Jan 2017_

On my other old VM which is CentOS 7, on this VM too MsQuic is working. On this VM when I run **ldconfig -p | grep libmsquic**, it shows following:
libmsquic.so.2 (libc6,x86-64) => /lib64/libmsquic.so.2
libmsquic.lttng.so.2.2.1 (libc6,x86-64) => /lib64/libmsquic.lttng.so.2.2.1
When I run **ldd /usr/lib/libmsquic.so.2**, it shows me this error **ldd: /usr/lib/libmsquic.so.2: No such file or directory**. 
Then I tried this **ldd /lib64/libmsquic.so.2**, it shows me followings;

linux-vdso.so.1 =>  (0x00007ffeac673000)
        libcrypto.so.1.1 => /lib64/libcrypto.so.1.1 (0x00007fdfa6fb3000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007fdfa6daf000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fdfa6b93000)
        libc.so.6 => /lib64/libc.so.6 (0x00007fdfa67c6000)
        libz.so.1 => /lib64/libz.so.1 (0x00007fdfa65b0000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fdfa7497000)

I have another question, this link https://github.com/microsoft/msquic/blob/main/docs/Platforms.md describes that libmsquic requires Openssl 1.1, however, on my VMs as you can see above, it is also working with OpenSSL 1.0.2k-fips, or when I install libmsquic, it also installs Opensll 1.1 itself, is it correct?

**Reply by ManickaP** [reply, marked as answer]
The `ldd` command needs to be run with your location of libmsquic, so in your case /lib64/..., my example used location on my machine.
As you can see the output of ldd, there is libcrypto 1.1 and that's what msquic needs. So it runs. The requirements mention OpenSSL 1.1, because that's the package that carries libcrypto 1.1.

And just FYI, CentOS 7 is rather old and will not be supported by .NET 8.
